shark-graph/src/main/java/shark/HprofHeapGraph.kt

```diff
@@ -103,13 +103,24 @@ class HprofHeapGraph internal constructor(
   // Pixel 2 XL API 28. Hit count was ~120K, miss count ~290K
   private val objectCache = LruCache<Long, ObjectRecord>(3000)
 
+  // java.lang.Object is the most accessed class in Heap, so we want to memorize a reference to it
+  private val javaLangObject: HeapClass
+
+  init {
+    val classId = index.classId("java.lang.Object")!!
+    val indexedObject = index.indexedObjectOrNull(classId)!!
+    javaLangObject = wrapIndexedObject(indexedObject, classId) as HeapClass
+  }
+
   override fun findObjectById(objectId: Long): HeapObject {
     return findObjectByIdOrNull(objectId) ?: throw IllegalArgumentException(
         "Object id $objectId not found in heap dump."
     )
   }
 
   override fun findObjectByIdOrNull(objectId: Long): HeapObject? {
+    if (objectId == javaLangObject.objectId) return javaLangObject
```

Comment:
Side note: we already have an LRU cache in front of operations that requires IO. However separately we also try to rely on the indexed objects data as much as possible to avoid as much IO as we can. What this PR does is skipping a ton of binary searches into the indexed objects when we look for the `java.lang.Object` class again and again (e.g. when going through class hierarchy for every class).

We could also add a small LruCache at this layer at some point. We'd have to evaluate the cost / benefit of adding a hash lookup to every `findObjectByIdOrNull` to avoid a binary search for a subset.